lib/hsa/mcwamp_hsa.cpp

```diff
@@ -299,6 +299,12 @@ class HSAQueue;
 class HSADevice;
 } // namespace Kalmar
 
+
+// Global trackers for all HSA agents:
+// Possible these should move as member static in the future.
+static std::vector<hsa_agent_t > g_hsaAgents; // Array of the CPU + all GPU agents. 
+static std::map<uint64_t, Kalmar::HSADevice *> g_agentToDeviceMap;
```

Comment:
Could `g_agentToDeviceMap` be a member variable of `HSAContext`? It can be then queries by `HSADevice` via a member function.